### ISSUE TITLE
Need advice on using this with locally cached app

### ISSUE THREAD

**Issue by Lagerberg** [Azure-Samples/active-directory-angularjs-singlepageapp#22]
Hi,
I am not sure if so many are listening here, but I believe my question might be of general interest:
I am writing a small app that needs to be working offline.
I have set up the caching to be done:
html lang="en-us"  ng-app="myApp" manifest="appcache.manifest"

The files are stored on an Azure web site, and my desire is to have some kind of authentication of the users. I have tried the Azure AD Web site authentication, to protect the site itself. That works pretty well, even if I still am unsure on how to identify the user in my app. The login via Azure etc. works fine, but then I guess there should be some code to save the returned token. I suspect ths token is being  sent to the Azure web site (which is not what I want, I need it locally in my app).

I went on to explore the ADAL JS functions, but after following the guides i get an error:
aadsts90002: No service namespace named was found in the data store.
I am not sure what it means..

In conclusion I'd like some advice o what the best method would be to be able to identify the Azure user, in an offline capable web app, hosted on Azure as a web site!

**Comment by dstrockis** [comment]
Hey @Lagerberg.  It seems to me like you're on the right track here.  You're correct - I wouldn't recommend using the web site authentication - you should integrate azure ad with your app as described in these guides.

I'd be happy to debug the AADSTS90002 error if you'd like... Can you provide a trace of the request you sent that resulted in this error?  I'd be curious to see where our guides led you astray.

Also, can you elaborate on what you mean by "offline capable"?  Is this app accessed through a browser?

**Comment by Lagerberg** [comment]
Hello dstrockis, very nice of you to respond!
I have done some reading up on this, and my understanding is that I should not enable any kind (site or mobile) authentication from my web app settings. I should register the app in azure ad, and also enable the implicit logon feature in the manifest file.
It looks like the adaljs library is very easy to use, and my error is probably just some kind of collision between the site authentication that I had enabled.
I'll get back with the results as soon as I have a chance to try it out (busy times!!)

Re your question about offline: Yes it's a browser based app.
The offline capable part is the fact that I have an application manifest statement:
html lang="en-us" ng-app="myApp" manifest="appcache.manifest"
in my index.html. That causes all the files listed in that file being downloaded and available locally, so I can run the app without a nextwork connection.

**Comment by Lagerberg** [comment]
Hi again,
I Think I found what was my main problem - in this part
adalAuthenticationServiceProvider.init(
        {
            // Config to specify endpoints and similar for your app
            tenant: "yyyyyyyy.onmicrosoft.com",              // Optional by default, it sends common
            clientId: "xxxx"

Fot Tenant: I should specify my Azure AD Tenant id, not my azure web app URL!
Also, the "common" default did not work in my case.

Thanks for the help!
//Peter

**Comment by dstrockis** [comment]
Nice! I didn't help much... :) 

Thanks for letting me know
